src/service/resources-impl.js

```diff
@@ -1359,13 +1362,13 @@ export class Resource {
      * Pending change height that was requested but could not be satisfied.
      * @private {number|undefined}
      */
-    this.pendingChangeHeight_;
-
-    /** @private {?function(*)} */
-    this.whenFirstLayoutCompleteResolve_ = null;
+    this.pendingChangeHeight_ = undefined;
 
-    /** @private @const {?Promise} */
-    this.whenFirstLayoutCompletePromise_ = null;
+    /** @private @const {!Promise} */
+    this.loadPromise_ = new Promise(resolve => {
```

Comment:
It leads to an ultra complicated state machine (which the original code messed up). I think here it is worth it, unless @dvoytenko feels differently.